coverage.py

```diff
@@ -26,11 +28,16 @@ class CoverageFailure(object):
 
     def __init__(self, provider, obj, exception, transient=True):
         self.obj = obj
-        self.output_source = provider.output_source
+        self.output_source = getattr(provider, 'output_source', None)
```

Comment:
WorkCoverageProvider doesn't have an output_source; this lets CoverageFailure work with both CoverageProvider and WorkCoverageProvider.